src/Yaapii.Atoms/Scalar/And.cs

```diff
@@ -2,147 +2,133 @@
-// copies or substantial portions of the Software.
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 //
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-// SOFTWARE.
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+// WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 using System;
 using System.Collections.Generic;
-using Yaapii.Atoms.List;
-using Yaapii.Atoms.Func;
-using Yaapii.Atoms.Scalar;
 using Yaapii.Atoms.Enumerable;
+using Yaapii.Atoms.Func;
 
 namespace Yaapii.Atoms.Scalar
 {
-    /// <summary>
-    /// Logical and. Returns true if all contents return true.
-    /// </summary>
+    /// <summary> Logical and. Returns true if all contents return true. </summary>
     /// <typeparam name="In"></typeparam>
     public sealed class And<In> : IScalar<Boolean>
     {
-        private readonly IEnumerable<IScalar<Boolean>> _enumerable;
-
-        /// <summary>
-        /// Logical and. Returns true after calling <see cref="IAction{X}"/> for every element.
-        /// </summary>
-        /// <param name="proc">the condition to apply</param>
-        /// <param name="src">list of items</param>
-        public And(Action<In> proc, params In[] src) : this(new ActionOf<In>(input => proc.Invoke(input)), src)
-        { }
-
-        /// <summary>
-        /// Logical and. Returns true after calling <see cref="IAction{X}"/> for every element.
-        /// </summary>
-        /// <param name="proc">the condition to apply</param>
-        /// <param name="src">list of items</param>
-        public And(Action<In> proc, IEnumerable<In> src) : this(new ActionOf<In>(ipt => proc.Invoke(ipt)), src)
-        { }
-
-        /// <summary>
-        /// Logical and. Returns true after calling <see cref="IAction{X}"/> for every element.
-        /// </summary>
-        /// <param name="proc">the condition to apply</param>
-        /// <param name="src">list of items</param>
-        public And(IAction<In> proc, params In[] src) : this(
-            proc, new EnumerableOf<In>(src))
-        { }
+        private readonly And _and;
```

Comment:
Using the non generic And is possible, but I think it is a better solution to use an enumerable.

Reason: This only works now, because the generic IEnumerable also implements the non-generic IEnumerable interface. This is designed this way by Microsoft, but in our world we don't do this. I think its clearer to use IEnumerable<In> inside.

Reply:
I dont agree.
Non generic IEnumerable is not used, dont get ur argument...
In my opinion there is no problem to use another class instead of duplicate code...

Reply:
Line 63 uses implicit type casting.

Reply:
??? So what, already changed but still dont get ur point.